src/lsp/cobol_parser/grammar.mly

```diff
@@ -3384,18 +3384,19 @@ let reversed_or_no_rewind_opt :=
 (* TODO: COB85 first format also takes instructions... *)
 
 %public let unconditional_action := ~ = perform_statement; < >
-let perform_statement :=
+let perform_statement [@context perform_stmt] :=
+  | PERFORM; po = ro(perform_phrase); isl = imp_stmts; END_PERFORM;
```

Comment:
Please reorder the two rules to have a diff focused on the essential changes: only the line starting with `let perform_statement` appears to need modification.

Reply:
I actually exchanged 
```ocaml
| PERFORM; po = ro(perform_phrase); isl = imp_stmts; END_PERFORM; [...]
```
and 
```ocaml
| PERFORM; i = procedure_name; [...]
```
in purpose in order to prevent `FOREVER` to be parsed as a procedure name in `PERFORM FOREVER`.

Reply:
The sequencing of production rules in such a grammar specification does not impact how a token is parsed: no precedence is derived from this order (apart maybe from cases of conflicts that are resolved arbitrarily — but we don't allow such a case to occur).